app.py

```diff
@@ -35,7 +38,18 @@ def index():
 
 @app.route("/bookmarks.html")
 def bookmarks():
```

Comment:
in between the lines with '@app.route' and 'def bookmarks()', add a line that just says '@is_logged_in'. This line will use the wrapper function to prevent anyone from accessing the bookmarks URL when they aren't logged in.

Reply:
"@app.route("/bookmarks.html")
@is_logged_in
def bookmarks():"

Is causing the website not to load at all and says "TypeError: exceptions must derive from BaseException"
including "self._flush_bg_loading_exception()" specifically.
I did it the same way as observed in the login function to make sure I wasn't causing the error in some way